src/kontrol/solc_to_k.py

```diff
@@ -66,6 +66,70 @@ def solc_to_k(
 
 @dataclass
 class Contract:
+    @dataclass
+    class Constructor:
+        sort: KSort
+        arg_names: tuple[str, ...]
+        arg_types: tuple[str, ...]
+        contract_name: str
+        contract_digest: str
+        contract_storage_digest: str
+        payable: bool
+        signature: str
+
+        def __init__(
+            self,
+            abi: dict,
+            contract_name: str,
+            contract_digest: str,
+            contract_storage_digest: str,
+            sort: KSort,
+        ) -> None:
+            self.signature = f'{contract_name}_init'
+            self.arg_names = tuple([f'V{i}_{input["name"].replace("-", "_")}' for i, input in enumerate(abi['inputs'])])
+            self.arg_types = tuple([input['type'] for input in abi['inputs']])
+            self.contract_name = contract_name
+            self.contract_digest = contract_digest
+            self.contract_storage_digest = contract_storage_digest
+            self.sort = sort
+            # TODO: Check that we're handling all state mutability cases
```

Comment:
I understand it's copied from a similar `Method` class, but should we address or remove this `TODO`? Other state mutability cases are `pure`, `view`, and `nonpayable`, and I'm not sure if we want to handle them differently. It doesn't have to be addressed in this PR though.

Reply:
I think `nonpayable` is handled implicitly by it not being payable, therefore we return a callvalue with a concrete value of zero. But I'm not aware of what we want to do with `pure` and `view`, or if we want to do something. Are those modifiers just something that is enforced by the solidity compiler or is it something we could take advantage of in verification?

This comment was originally added in https://github.com/runtimeverification/evm-semantics/pull/1318.